### PR TITLE
chore(ci): remove v1 workflows

### LINKED ISSUE #1616: Maintenance: remove v1 workflows from v2 before merge
### Summary

We should remove the v1 workflows from the code before merging the v2 code.

### Why is this needed?

When we started creating specific workflows for v2, we created separate files. We should now remove the v1 workflows, so they won't interfere with the release process of v2.

### Which area does this relate to?

Automation

### Solution

_No response_

### Acknowledgment

- [X] This request meets [Lambda Powertools Tenets](https://awslabs.github.io/aws-lambda-powertools-python/latest/#tenets)
- [ ] Should this be considered in other Lambda Powertools languages? i.e. [Java](https://github.com/awslabs/aws-lambda-powertools-java/), [TypeScript](https://github.com/awslabs/aws-lambda-powertools-typescript/)

### PR DESCRIPTION
**Pull request by rubenfonseca**
**Issue number:** #1616

### Changes

> Please provide a summary of what's being changed

Removes workflows related to creating the v1 layer, which we will not use on v2.
Removes dangerous workflows from v2 that would overite commits in develop.
Switches to workflow_calls on the laye workflow.

### User experience

> Please share what the user experience looks like before and after this change

This prepares us to merge v2 into develop, and not call the old workflows when releasing a new version.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [ ] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: .github/workflows/publish_v2_layer.yml [hunk 1]
```diff
@@ -10,10 +10,10 @@ on:
...
         description: "Latest PyPi published version to rebuild latest docs for, e.g. v2.0.0"
         required: true
-  # workflow_run:
-  #   workflows: ["Publish to PyPi"]
-  #   types:
-  #     - completed
+  workflow_run:
+    workflows: ["Publish to PyPi"]
+    types:
+      - completed
```
Comment (rubenfonseca) [inline .github/workflows/publish_v2_layer.yml]: Instead of `workflow_run`, use a `workflow_call` and call it explicitly after the "Publish to Pypi" action

**Comment by rubenfonseca** [comment]
On `.github/workflows/on_release_notes.yml`, filter the `on: release` trigger with the correct tag prefix

**Review by leandrodamascena — APPROVED** [review]
LGTM

**Comment by heitorlessa** [comment]
Looking...

**Comment by heitorlessa** [comment]
LGTM except one thing - we're gonna need a Post-Launch list to automate V1 releases. For this PR, the `if startsWith` correctly limits Release Notes for V2 only - we'll need a `workflow_dispatch` to release V1.

That is:

* Make changes to V1 -> Merge into `v1`
* Use Workflow X to release V1.x --> PyPi, Lambda Layer, (SAR?)
* Publish a new release notes for V1.x

**Review by heitorlessa — APPROVED** [review]

**Comment by rubenfonseca** [comment]
> LGTM except one thing - we're gonna need a Post-Launch list to automate V1 releases. For this PR, the `if startsWith` correctly limits Release Notes for V2 only - we'll need a `workflow_dispatch` to release V1.
> 
> That is:
> 
> * Make changes to V1 -> Merge into `v1`
> * Use Workflow X to release V1.x --> PyPi, Lambda Layer, (SAR?)
> * Publish a new release notes for V1.x

Thanks @heitorlessa tracking this here https://github.com/awslabs/aws-lambda-powertools-python/issues/1618

**Pull request merged**